### PR TITLE
Initial commit for adding images

### PR DESCRIPTION
**Pull request by ahmelsayed**
Signed-off-by: Ahmed ElSayed <ahmels@microsoft.com>

/cc @zroubalik @tomkerkhove 

Not sure if this is what you sort of had in mind for this repo. I created a Dockerfile for all the images we are using in the e2e folder in the KEDA repo. most of them are just an exact reference to the current image used. I only added the code and actual Dockerfile for the one I did.

### CODE REVIEW

File: e2e/images/build.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#! /bin/bash
+set -e
+
+DIR=$(dirname "$0")
+cd $DIR
+
+function push {
+    for IMAGE_NAME in $(find * -name Dockerfile -exec dirname {} \; | tr '/' '-')
+    do
+        docker push docker.io/kedacore/$IMAGE_NAME:latest
```
Comment (ahmelsayed) [inline e2e/images/build.sh:10]: I'm not sure about the naming for these images. this change currently assumes that any folder in `e2e/images` that contains a `Dockerfile` is an "image". So for example

```
├── artemis
│   ├── consumer
│   │   └── Dockerfile
│   └── producer
│       └── Dockerfile
├── azure-blob
│   └── Dockerfile
├── kafka
│   └── Dockerfile
```

results in 

```
docker.io/kedacore/artemis-consumer
docker.io/kedacore/artemis-producer
docker.io/kedacore/azure-blob
docker.io/kedacore/kafka
```


I don't like polluting the top level repo there, but not sure what's the best alternative.

File: e2e/images/prometheus/Dockerfile [hunk 2]
```diff
@@ -0,0 +1 @@
+FROM tbickford/simple-web-app-prometheus@sha256:d11e96a7b5cb90d1d60abc10f073411d9fdbf5073e58041b4f2e1055d16e16b7
```
Comment (tomkerkhove) [inline e2e/images/prometheus/Dockerfile:1]: I think @tbickford was willing to donate this one, but I'm not sure

File: e2e/images/rabbitmq/Dockerfile [hunk 3]
```diff
@@ -0,0 +1 @@
+FROM jeffhollan/rabbitmq-client@sha256:c5cd232389260e9841ced013676ad7f40ee76cd73a14cb6a46b90c59a5a2f3cd
```
Comment (tomkerkhove) [inline e2e/images/rabbitmq/Dockerfile:1]: We should ask @jeffhollan if we can move it here or clone it maybe?

File: e2e/images/kafka/Dockerfile [hunk 4]
```diff
@@ -0,0 +1 @@
+FROM jeffhollan/twitter-function@sha256:7494a28aa99ea76ac73c312776982d445f20c557d123a66e8be732f2c7163d60
```
Comment (tomkerkhove) [inline e2e/images/kafka/Dockerfile:1]: We should ask @jeffhollan if we can move it here or clone it maybe?

File: e2e/images/build.sh (hunk 1 above)
Comment (tomkerkhove) [reply e2e/images/build.sh:10]: LGTM

Comment (tomkerkhove) [reply e2e/images/build.sh:10]: But latest is very hard to troubleshoot, can we use commit or version tag instead?

Comment (zroubalik) [reply e2e/images/build.sh:10]: LGTM, maybe add `tests-` or `e2e-` prefix to the image names?

File: e2e/images/redis/consumer/Dockerfile [hunk 5]
```diff
@@ -0,0 +1 @@
+FROM abhirockzz/redis-streams-consumer@sha256:6cb26d184045d9bce495324a82ae2fda6ba4d52115497a0d331f990fbc8da0c7
```
Comment (zroubalik) [inline e2e/images/redis/consumer/Dockerfile:1]: We could ask @abhirockzz for the sources of this image

File: e2e/images/redis/producer/Dockerfile [hunk 6]
```diff
@@ -0,0 +1 @@
+FROM abhirockzz/redis-streams-producer@sha256:79a6dda04b1a54c4ddd1e802ac8447b9c802adf9860cba8cf65017a3e333e7fa
```
Comment (zroubalik) [inline e2e/images/redis/producer/Dockerfile:1]: We could ask @abhirockzz for the sources of this image

File: e2e/images/artemis/consumer/Dockerfile [hunk 7]
```diff
@@ -0,0 +1 @@
+FROM balchu/kedartemis-consumer@sha256:c219ffcdc0a464ccecb0d7d9d900beace2e739b6510227492c0250715af2af8d
```
Comment (zroubalik) [inline e2e/images/artemis/consumer/Dockerfile:1]: @balchua are you willing to contribute the sources in here?

File: e2e/images/artemis/producer/Dockerfile [hunk 8]
```diff
@@ -0,0 +1 @@
+FROM balchu/artemis-producer@sha256:6a4b95188bccd7b0ae387950cb28ef7ad60f085b871298161afcdece0d634b35
```
Comment (zroubalik) [inline e2e/images/artemis/producer/Dockerfile:1]: @balchua are you willing to contribute the sources in here?

File: e2e/images/build.sh (hunk 1 above)
Comment (zroubalik) [reply e2e/images/build.sh:10]: @tomkerkhove I don't expect big changes in those images and they are used in one place, so `latest` is not a big deal. But adding a version on anything could do any harm, right? 😄

Comment (ahmelsayed) [reply e2e/images/build.sh]: I made the abbreviated commitid be the tag for the images.

File: e2e/images/artemis/consumer/Dockerfile (hunk 7 above)
Comment (balchua) [reply e2e/images/artemis/consumer/Dockerfile:1]: Yup. No problem at all.

File: e2e/images/artemis/producer/Dockerfile (hunk 8 above)
Comment (balchua) [reply e2e/images/artemis/producer/Dockerfile:1]: Yup no problem at all

File: e2e/images/prometheus/Dockerfile (hunk 2 above)
Comment (tbickford) [reply e2e/images/prometheus/Dockerfile:1]: Yep! I'll be working on this at some point later this week.

**Comment by zroubalik** [comment]
@ahmelsayed for some reason the docker push returns `anuthorized` error, even though `docker login` succeeded in previous step. I was able to push those images from my local machine though 🙄

File: e2e/images/redis/producer/Dockerfile (hunk 6 above)
Comment (abhirockzz) [reply e2e/images/redis/producer/Dockerfile:1]: sure.. will do.

File: e2e/images/redis/consumer/Dockerfile (hunk 5 above)
Comment (abhirockzz) [reply e2e/images/redis/consumer/Dockerfile:1]: sure.. will do.

**Comment by ahmelsayed** [comment]
looks like the ci user needs explicit permissions for every repo on kedacore. I forgot I set it up that way originally. it also makes that the process of adding new images is a bit complicated as some other user needs to create the repo, give ci user access and then merge the PR. 

The only other role I see in docker hub is an org owner. I think I was trying to avoid making the ci user an org owner, but it might be the only way.

**Pull request merged**